lib/store_attribute/active_record/type/typed_store.rb

```diff
@@ -84,7 +98,16 @@ def type_for(key)
         accessor_types.fetch(key.to_s)
       end
 
-      attr_reader :accessor_types, :store_accessor
+      def has_default?(key)
+        defaults.key?(key.to_s)
+      end
+
+      def get_default(key)
+        value = defaults.fetch(key.to_s)
+        value.respond_to?(:call) ? value.call : value
```

Comment:
Checking for Proc is cheaper than `respond_to?` (see also how this is implemented in Active Model https://github.com/rails/rails/blob/b2eb1d1c55a59fee1e6c4cba7030d8ceb524267c/activemodel/lib/active_model/attribute/user_provided_default.rb#L7)

```suggestion
        value.is_a?(Proc) ? value.call : value
```